### PR TITLE
JUCX: remove c++11 requirement.

### PR DESCRIPTION
**Pull request by petro-rudenko**
## What
Remove requirement for c++11, since for now not used any features of c++11. 

## Why ?
To compile on old gcc.  cc @alinask

### CODE REVIEW

**Comment by alinask** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9891/ for details (Mellanox internal link).

**Comment by alinask** [comment]
@yosefe Can you please review?

File: bindings/java/src/main/native/Makefile.am [hunk 1]
```diff
@@ -43,7 +43,7 @@ libjucx_la_SOURCES = context.cc \
                      ucs_constants.cc \
                      worker.cc
 
-libjucx_la_CXXFLAGS = -fPIC -DPIC -Werror -std=c++11
+libjucx_la_CXXFLAGS = -fPIC -DPIC -Werror
```
Comment (yosefe) [inline bindings/java/src/main/native/Makefile.am]: BTW, shall we add BASE_CXXFLAGS here, which also have -O3 -Wall -g (could be another PR)
